pkg/pubsub/gcp/gcp.go

```diff
@@ -0,0 +1,69 @@
+// Package gcp implements the interface with Google Cloud Pub/Sub.
+package gcp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"google.golang.org/api/option"
+)
+
+const URIIdentifier = "gcppubsub"
+
+var (
+	re = regexp.MustCompile(`^gcppubsub://projects/([^/]+)/locations/[^/]+/topics/([^/]+)$`)
+)
+
+type Publisher struct {
+	client *pubsub.Client
+	topic  string
+}
+
+func New(ctx context.Context, refStr string, opts ...option.ClientOption) (*Publisher, error) {
+	projectID, topic, err := parseRef(refStr)
+	if err != nil {
+		return nil, fmt.Errorf("parse ref: %w", err)
+	}
+	client, err := pubsub.NewClient(ctx, projectID, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("create pubsub client for project %q: %w", projectID, err)
+	}
+	return &Publisher{
+		client: client,
+		topic:  topic,
+	}, nil
+}
+
+func (p *Publisher) Publish(ctx context.Context, entry models.LogEntryAnon) error {
+	data, err := json.Marshal(entry)
```

Comment:
instead of JSON marshalling `models.LogEntryAnon` directly, we should really convert this into the Sigstore bundle format using the two methods in https://github.com/sigstore/rekor/blob/main/pkg/tle/tle.go that can convert `models.LogEntryAnon` into the bundle format JSON.

Reply:
We had also talked about potentially defining a https://cloudevents.io/ schema for these messages, which would be nice to do from the start. Thoughts?

Reply:
Ack re using the TLE methods.

I wasn't aware of the Cloud Events spec, but it seems reasonable to follow. I had planned on adding some of these anyway in a future PR, I can tackle that then if it is OK with you. I prefer to keep individual PRs smaller and easier to review, atomically roll-back if needed, etc.

Reply:
@bobcallaway I have defined a proto for this in https://github.com/sigstore/protobuf-specs/pull/85. I opted to use the JSON format [1] instead of the protobuf format [2] for the initial implementation for better compatibility with PubSub providers (ex AWS SNS) which require JSON for event filtering to work. We can consider also publishing the protobuf format to a second topic in the future if there's demand. Please take a look when you have time.

[1] https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/formats/json-format.md
[2] https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/formats/protobuf-format.md

Reply:
What do you think about reusing the existing `TransparencyLogEntry` type? This would make each pub/sub message larger, but would prevent diverging message contents and allow for verification of each message (since each message will contain an inclusion proof).

Reply:
Sorry, I missed that this would be encoded in `data`. Could we use the TLE type instead and annotate the type with JSON annotations if needed?

Reply:
@bobcallaway @haydentherapper I wasn't happy with the approach in this PR, so I have rewritten it on a new branch and opened a separate PR:  #1580. Apologies for the churn, but I think the new approach is much better.